PerfNext/app/apis/BenchEngine/parser.js

```diff
@@ -338,83 +292,157 @@ function parseSetupScript(raw_definition, callback){
+			let iterations = result.benchmarks.benchmark.iterations;
 
-			var generatedOutput = generateDate();
-			generatedOutput += eval('`'+header+'`'); //Evaluate header params (${XXXXXX})
-			generatedOutput += '\n######### Generated Script #########\n';
+			// Changes to this delimiter affect TestResultSummaryService's BenchmarkParser class.
+			// Please confirm any delimiter changes
+			let benchmarkJobDelimiter = "********** START OF NEW TESTCI BENCHMARK JOB **********";
 
-			if(iterations > 1){
-				generatedOutput += "iteration=0 \n";
-				generatedOutput += `while [ "$iteration" -lt ${iterations} ]` + '\n';
-				generatedOutput += 'do \n';
-				generatedOutput += "echo \"Start of iteration $iteration\" \n";
-			}
+			getHeader(function (header){
 
-			generatedOutput += `echo ""\n`;
-			generatedOutput += `echo "${benchmarkJobDelimiter}"\n`;
-			generatedOutput += `echo "Benchmark Name: ${benchmarkName} Benchmark Variant: ${benchmarkVariant}"\n`;
-			generatedOutput += `echo "Benchmark Product: ${buildName}"\n`;
-			generatedOutput += `echo ""\n`;
-			generatedOutput += generateENV(result);
-			generatedOutput += '\n## HW Specific Environment Vars ##\n';
-			generatedOutput += generateHWENV(result);
-			generatedOutput += '\n';
+				var generatedOutput = generateDate();
+				generatedOutput += eval('`'+header+'`'); //Evaluate header params (${XXXXXX})
+				generatedOutput += '\n######### Generated Script #########\n';
+
+				generatedOutput += `echo "iterations = ${iterations}"\n`;
+
+				generatedOutput += `echo ""\n`;
+				generatedOutput += `echo "${benchmarkJobDelimiter}"\n`;
+				generatedOutput += `echo "Benchmark Name: ${benchmarkName} Benchmark Variant: ${benchmarkVariant}"\n`;
+				generatedOutput += `echo "Benchmark Product: ${buildName}"\n`;
+				generatedOutput += `echo ""\n`;
+				generatedOutput += generateENV(result);
+				generatedOutput += '\n## HW Specific Environment Vars ##\n';
+				generatedOutput += generateHWENV(result);
+				generatedOutput += '\n';
 			
-			if (scriptType == 'java')
-			{
-				generatedOutput += generateJavaRunCommands(result);
-			}
-			else
-			{
-				generatedOutput += generateScriptCommands(result);
-			}			
+				if (scriptType == 'java')
+				{
+					generatedOutput += generateJavaRunCommands(result);
+				}
+				else
+				{
+					generatedOutput += generateScriptCommands(result);
+				}			
+
+				fs.readFile('logo.txt', 'utf8', function(err, contents) {
+					generatedOutput = contents + '\n' +  generatedOutput;
+
+					callback(generatedOutput);
+				});
+			});
+		});
+	}
+}
+// Submit benchmark run request to jenkins
+function jenkinsSubmit(generatedOutput, generatedOutputBaseline, setupScript, setupScriptBaseline, req, res) {
+	var setupMachine=req.query.machine;
+	console.log('parser.js: setupMachine: '+setupMachine);
+	
+	// Parse iteration number from benchmark script
+	var iterationStart = generatedOutput.indexOf("iterations = ") + "iterations = ".length;
```

Comment:
Method to get iteration looks a bit hacky. We can come back to it later in order to prevent this PR from merging.